### ISSUE TITLE
How does "Code time by genre" work?

### ISSUE THREAD

**Issue by darkvertex** [swdotcom/swdc-vscode#9]
Hey guys,

On the web dashboard I saw the "Code time by genre", not with actual data, but as a suggested graph that may eventually appear.

I couldn't find any docs about this feature. Is it just in the roadmap and not publicly available yet?

I think it's a very interesting thing to track. :thinking:

**Comment by andrewhli** [comment]
Hi Alan. Appreciate your feedback. The music by genre is a future release that we're currently working on.

**Comment by darkvertex** [comment]
Ah, cool. Looking forward to it!

If I may ask, how do you intend to source the data? Access _scrobbled_ tracks from last.fm/musicbrainz and correlate that? Or direct tap to a few generic music players?

**Comment by andrewhli** [comment]
Hi there. We are planning to integrate with Spotify and iTunes.

**Comment by darkvertex** [comment]
Hey, that's cool!

If I may make a suggestion (as I'm more of a Google Play Music user), one easy'ish way to support Google as well, if you wanted, would be to tap into the unofficial player project here:
https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-
which has two local APIs for controlling and listening to events from the player. Their [WebSocket API is great for streaming](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/blob/master/docs/PlaybackAPI_WebSocket.md) "now playing" events.

There's even [a vscode extension already made that shows you what's playing in your statusbar](https://github.com/Jodything/vsCodeGoogleMusic) using this unofficial player.

And obviously another interesting way to support a lot of things, not just Google, would be to tap into the user's last.fm account and [use APIs](https://www.last.fm/api) to correlate their "now-playing history" of plays with whatever "coding time" events you store on your end. This is assuming a user is using one of dozens of "[scrobblers](https://www.last.fm/about/trackmymusic)" (song play trackers) that ping last.fm as you listen to things.

Just my $0.02, keep up the good work. :+1:

**Comment by blackjack4494** [comment]
Why don't make use of vscode spotify plugin? or make a web app to keep track of what a user is listening to?

**Comment by andrewhli** [comment]
Both good ideas. I'm wondering what the best way to integrate with the VS Code Spotify plugin would be. Let me know if you have any ideas there. Would love to make a web app component for this that works on app.software.com. In your view, what should a web app show?

**Comment by mmclead** [comment]
Music Time was released this year!  https://www.software.com/music-time

Thank you @darkvertex for the details on Google Play API's and Last.Fm

**Comment by darkvertex** [comment]
@mmclead I'm a bit bummed that it's only supporting Spotify. I'm on Youtube Music (formerly Google Play Music) now and I can't enjoy this neat feature. :/ Congrats on the release in any case!

Again, I think tapping to Last.fm's "scrobbles" would be a good route as it could let you piggyback on their support of song trackers in multiple music services and platforms (which include Spotify, even!)

**Comment by mmclead** [comment]
I hear you.  I'm an avid sublime text user and music time isn't on sublime yet either.  

We do have expansion of supported streaming services and editors on the roadmap.  We had to make a choice about which to support and Spotify has an official API that we can count on the be stable and available, so we went with that for the first launch.  

As we build more I'm taking a serious look at your suggestions as they seem like really good options. 

Thanks for the research done on that and we hope to support your setup soon.
